src/Core/ConfigurationDefaults.re

```diff
@@ -36,6 +36,12 @@ let getDefaultConfigString = configName =>
   "workbench.sideBar.location": "left",
   "workbench.sideBar.visible": true,
   "workbench.statusBar.visible": true,
+  "workbench.statusBar.items": {
+    "left": ["notificationCount", "macro", "leftItems", "diagnosticCount", "git"],
```

Comment:
I noticed in the demo from @bryphe below that we need to adjust the default configuration shipped in Oni to eliminate the spacing gap between notification items by default. It's caused by items hidden due to `showOnNotificaiton` still occupying their position.

To remove the spacing we can adjust the default configuration to something like this.

```json5
{
  "workbench.statusBar.items": {
    "left": ["modeIndicator", "notificationCount", "diagnosticCount", "notification", "..."],
    "right": ["...", "modeIndicator"],
    "showOnNotification": ["notificationCount", "modeIndiactor", "diagnosticCount"]
  }
}
```

Reply:
Actually that's not necessary, but it might be a good idea, to make 'diagnosticCount' be next to the `notificationCount` and always the shown.

By default, `diagnosticCount` is not on `showOnNotification` so it will be hidden.

Also, I changed the code now so that the `default` behavior is for the `showOnNotification` to be forced grouped together on their side, ex:
```JSON
  "start": ["aa", "bb", "cc"]
  "showOnNotification": ["aa", "cc"]
```
Then it's shown as

```JSON
["aa", "cc", "bb"]
```
And then I added a new `noficationMode` `keepPosition` where it forces the items to be in the position that user specified, but it will cause those gaps

Reply:
Not sure I follow. I thought `diagnosticCount`  is part of the default notification pop. If it's not then nevermind

But the point I was trying to make is to explicitly define the `notificationMessage` position next to the items that will be shown in `items.start`. That means something like below. That ensures by default there is no gap. If the user changed the ordering then they need to make it a way that doesn't introduce gap as well:

```
{
  "workbench.statusBar.items": {
    "left": ["notificationCount", "notificationMessage", "..."],
    "right": ["...", "modeIndicator"],
    "showOnNotification": ["notificationCount", "modeIndiactor"]
  }
}
```

> Also, I changed the code now so that the default behavior is for the showOnNotification to be forced grouped together on their side

That can also be a fix for the issue. But the reason I didn't suggest that because it _rearranges_ some of the elements on the status bar only to revert back to the original positioning a second later, I think that can result in a bad UX. But I didn't see it in practice, maybe it's not that bad.

Reply:
> That can also be a fix for the issue. But the reason I didn't suggest that because it rearranges some of the elements on the status bar only to revert back to the original positioning a second later, I think that can result in a bad UX. But I didn't see it in practice, maybe it's not that bad.

Sorry I have seamed to not explain my self correctly...

It will be shown as this all the time not only for the notification

```JSON
["aa", "cc", "bb"]
```

Reply:
Oh, I see, I misunderstood. If I get it correctly now, it means we will automatically group notifications even though the user might have chosen a different positioning. To force the items to be kept in the same position as the configs the `keepPosition` mode must be set, right?

I feel like that defeats the purpose of explicitly positioning the teams via configs... in general, I'd try to make things [Simple and Stupid](https://en.wikipedia.org/wiki/KISS_principle). In this case, It seems like we are adding complexity for a little gain especially since the fix for gapping can be done by the user him/herself with little effort. We just need to make our defaults gapless.

But I also feel like I'm being too picky here, sorry. so I'd leave it for @bryphe to decide what makes sense. Thanks for the great work on this 🙇🏻‍♂️

Reply:
| Oh, I see, I misunderstood. If I get it correctly now, it means we will automatically group notification items even though the user might have chosen a different positioning. To force the items to be kept in the same position as the configs the keepPosition mode must be set, right?

Yes

> I feel like that defeats the purpose of explicitly positioning the teams via configs... in general, I'd try to make things Simple and Stupid. In this case, It seems like we are adding complexity for a little gain especially since the fix for gapping can be done by the user him/herself with little effort. We just need to make our defaults gapless.

True, true then maybe what we have to do it's have default to match what the user put and then maybe have a `prettify` option tries to move everything together? What do you think?